### ISSUE TITLE
Agile bounding boxes for layers as new features are added/appended

### ISSUE THREAD

**Issue by jonpmarino** [MapStory/mapstory#439]
This is a spike...research needs to be done first before solutions considered or proposed in a future cycle. Just plotting the thought now.

Scenario:

As we incorporate versioned editing into layers more and more - either layers that are imported using importer, or ones that are created using "create layer" - we will increasingly encounter the issue of layers needing new bounding boxes that are reflective of new data added to the layer.

So, for example, right now if data is imported with layer title "Pubs of England", but the data itself only has pubs located in London, then the bounding box for the layer will be zoomed into London, not England, since that's what the data shows. But, over time if other users add more pubs throughout england (either one at a time or in bulk with append) the layer will gradually encompass a larger geography, and the default bounding box view for the layer should expand to reflect those added features.

This is especially relevant for layers created with Create Layer. Created Layers have no features to start with. The idea is that a user creates an empty schema and then through versioned editing can add features to the layer (a great tool for people that just _know_ alot about a topic but don't have the pre-existing geodata for it to upload). Right now there is nothing in the Create Layer workflow that generates a bounding box view that is reflective of the data in the layer. All layers are max zoomed out (Issue #365 will at least help with this a bit). As features are added to a layer, it would make sense to have the default zoom reflect those features' geographic bounds. For example, if all features are added in the same city, it doesn't make sense to have the layer zoom out to a global view but instead in on that city. Then, if more features are added in far away lands (i.e. other continents) it makes sense for the geographic bounds for the layer to expand.

**Comment by jonpmarino** [comment]
@smithkm is working this through #353. May be able to close this. Up to @smithkm

**Comment by smithkm** [comment]
Yes the plugin to do this is done: https://github.com/smithkm/ms-gs-plugins

It just needs to be incorporated into Mapstory.

**Comment by jonpmarino** [comment]
kevin do we need a new ticket for work to incorporate into mapstory?

**Comment by smithkm** [comment]
Needed to deploy: @cuttlefish 

- [ ] Repo needs to be moved to MapStory github account
- [ ] JAR needs to be built and injected into our GeoServer WAR

Nice to have: @smithkm

- [ ] copy the test infrastructure from https://github.com/boundlessgeo/geoserver-exts to make future development of plugins easier.

**Comment by jonpmarino** [comment]
@smithkm @cuttlefish  what is the status of this for MapStory?

I just encountered a good test case. I uploaded a layer to beta.mapstory.org showing breweries in Wisconsin. 2 of the features were missing a negative sign on their Lon so ended up in Central Asia instead of Wisconsin. As a result, the bounding box is over Europe - the geographic center of my points, although all but 2 are actually in Wisconsin. With versioned editing I fixed this. Now all points are in Wisconsin. But the bounding box for the layer has not changed...still appears over Europe. Thus, when a user hits play on the layer, they don't see any features unless they know to scroll to Wisconsin.

Link to referenced layer: https://beta.mapstory.org/layers/geonode:beer_breweries___breweries_21e31ce2
